contributors/design-proposals/node/node-fence.md

```diff
@@ -0,0 +1,528 @@
+apiVersion: 1.0
+metadata:
+  - name fence_method_template
+  - version: 1.0
+data:
+  - method_template.properties: |
+name=eaton_pdu
+agent_name=fence_eaton_snmp
+must_sucess=yes
+ipaddr=192.168.1.4
+password=eaton_password
+username=eaton_admin
+snmp-priv-prot=AES
+snmp-priv-passwd=eaton_snmp_passwd
+snmp-sec-level=authPriv
+inet4-only
+```
+
+Method for enable and disable of brocade for node A:
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+ - method.properties: |
+name=nodeA_fc_off
+template=fc_switch_brocade
+plug=1
+```
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+ - method.properties: |
+name=nodeA_fc_on
+template=fc_switch_brocade
+plug=1
+action=on
+```
+
+Method for enable and disable of brocade for node B:
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+spec:
+ - method.properties: |
+name=nodeB_fc_on
+template=fc_switch_brocade
+plug=2
+```
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+spec:
+ - method.properties: |
+name=nodeB_fc_off
+template=fc_switch_brocade
+plug=2
+action=on
+```
+
+Methods for on/off APC (both nodes):
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+ - method.properties: |
+name=nodeA_apc_off
+template=apc_pdu
+plug=1
+action=off
+```
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+ - method.properties: |
+name=nodeB_apc_off
+template=apc_pdu
+plug=2
+action=off
+```
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+ - method.properties: |
+  - name=nodeA_apc_on
+  - template: apc_pdu
+  - parameters:
+     - plug: 1
+     - action: on
+```
+
+```
+kind: FenceMethodConfig
+apiVersion: 1.0
+metadata:
+  - version: 1.0
+spec:
+  - Name: NodeB_apc_on
+  - template: apc_pdu
+  - parameters:
+     - plug: 2
+     - action: on
+```
+
+Method for on/off eaton (both nodes)
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+ - method.properties: |
+Name=nodeA_eaton_off
+template=eaton_pdu
+plug=1
+action=off
+```
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+ - method.properties: |
+name=NodeA_eaton_off  
+template=eaton_pdu  
+plug=2
+action=off
+```
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_method_config
+  - version: 1.0
+data:
+  - method.properties: |
+name=nodeA_eaton_on
+template=eaton_pdu 
+plug=1
+action=on
+```
+
+```
+kind: ConfigMap
+apiVersion: 1.0
+metadata:
+  - name: fence_mothod_config
+  - version: 1.0
+data:
+  - method.properties: |
+name=nodeA_eaton_on
+template=eaton_pdu  
+plug=2
+action=on
+```
+
+Finally fence configs:
```

Comment:
i feel like the api definition should encode intent and the means of reaching that intent should be abstracted from the user api.  right now, it feels like i need intimate knowledge of the config map(s) to know what to specify in a NodeFenceConfig which seems less than ideal to me.

Reply:
The controller and execution are quite intuitive. we need to store parameters for each node devices\apis, and admin must have intimate knowledge with those configmaps, each describes fencing device or api that need to be attached to the node.. do you suggest to use external store for describing the methods' fields?